crates/indexer/src/pool.rs

```diff
@@ -0,0 +1,24 @@
+use std::time::Duration;
+
+use sqlx::{
+    postgres::{PgConnectOptions, PgPoolOptions},
+    ConnectOptions, PgPool,
+};
+
+use crate::config::read_indexer_config;
+
+lazy_static::lazy_static! {
+    pub static ref POOL: PgPool = {
+        let indexer_config = read_indexer_config("./indexer-config.toml").unwrap();
+
+        smol::block_on(async {
+            let mut opts: PgConnectOptions = indexer_config.pg_url.parse()?;
+            opts.log_statements(log::LevelFilter::Debug)
+                .log_slow_statements(log::LevelFilter::Warn, Duration::from_secs(5));
+            PgPoolOptions::new()
+                .max_connections(5)
+                .connect_with(opts)
+                .await
+        }).unwrap()
```

Comment:
suggest to use `now_or_never` without runtime poll

```suggestion
        let mut opts: PgConnectOptions = indexer_config.pg_url.parse().expect("...");
        opts.log_statements(log::LevelFilter::Debug).log_slow_statements(log::LevelFilter::Warn, Duration::from_secs(5));

        PgPoolOptions::new().max_connections(5).connect_with(opts).now_or_never().expect("...")